### ISSUE TITLE
Pages broken after WP 3.14 and CCTM 0.9.1 upgrade

### ISSUE THREAD

**Issue by GoogleCodeExporter** [viktorthang/wordpress-custom-content-type-manager#108]
```
What steps will reproduce the problem?
1. Upgrade WP to 3.1.4
2. Upgrade CCTM to 0.9.1

What is the expected output? What do you see instead?
Pages stop working and cannot be found (404) when CCTM is activated. Deactivate 
the plugin and the pages start working again.


Does the problem continue if you disable all other plugins?
Yes, problem is definitely linked to CCTM.

I've tried upgrading the plugin to the latest 0.9.3 release, with no effect on 
the problem. Also tried installing 0.8.9 but this didn't fix the problem either

Please provide any additional information below (e.g. Browser, version,
OS):

```

Original issue reported on code.google.com by `supp...@switchomatic.co.uk` on 3 Jul 2011 at 11:39

**Comment by GoogleCodeExporter** [comment]
```
Have you tried dowgrading WP to 3.1.3 ? Or upgrading it to 3.2 RC ? 

I also have this issue. Will try the downgrade and if have success will let 
know.
```

Original comment by `lsant...@gmail.com` on 3 Jul 2011 at 9:01

**Comment by GoogleCodeExporter** [comment]
```
WP 3.1.3 + CCTM 0.9.1 , same issue.
WP 3.1.3 + CCTM 0.9.0 , works ok.


```

Original comment by `lsant...@gmail.com` on 3 Jul 2011 at 10:11

**Comment by GoogleCodeExporter** [comment]
```
There isn't full support for permalinks yet.  See related issue somewhere here 
in the bug-tracker.  WP does not make it easy to handle these unfortunately, so 
for a lot of these issues, it's be me against WP, and I've had to rewrite 
significant portions of the core to get what people would consider "normal" 
functionality.

I'd recommend using just the standard GET parameters for your custom post types 
and not any of the custom permalink stuff -- turn off all the permalink stuff 
and things should work fine. 

If that basic stuff isn't working, then it's probably due at least in part to 
your server/environment.  Let me know if the issue continues.
```

Original comment by `fireproofsocks` on 3 Jul 2011 at 10:33
- Changed state: **InReview**

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
I'm seeing exactly the same problem. WP3.1.3 + CCTM 0.9.0 does fine, 0.9.1 and 
above cause these errors:

Warning: Illegal offset type in isset or empty in 
/homepages/15/d372235597/htdocs/cg-com/wp-includes/post.php on line 817

817 is in this function:

function get_post_type_object( $post_type ) {
    global $wp_post_types;

    if ( empty($wp_post_types[$post_type]) )
        return null;

817>>   return $wp_post_types[$post_type];
}

AND 

Warning: Cannot modify header information - headers already sent by (output 
started at /homepages/15/d372235597/htdocs/cg-com/wp-includes/post.php:817) in 
/homepages/15/d372235597/htdocs/cg-com/wp-includes/pluggable.php on line 897

This error is part of the redirect system:

function wp_redirect($location, $status = 302) {
    global $is_IIS;

    $location = apply_filters('wp_redirect', $location, $status);
    $status = apply_filters('wp_redirect_status', $status, $location);

    if ( !$location ) // allows the wp_redirect filter to cancel a redirect
        return false;

    $location = wp_sanitize_redirect($location);

    if ( !$is_IIS && php_sapi_name() != 'cgi-fcgi' )
        status_header($status); // This causes problems on IIS and some FastCGI setups

897>>   header("Location: $location", true, $status);
}

I should also note that turning off permalinks doesn't help; the page redirects 
to the home page. Thanks.

```

Original comment by `david.ca...@gmail.com` on 4 Jul 2011 at 3:09

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
I have just downgraded Wordpress and CCTM to 3.1.3 and 0.9.0 respectively and 
this is still broken. 

I did only overwrite the files to perform the downgrade, rather than touch 
anything in the database. Is there anything in the DB that I need to change? 
Wordpress is still reporting 3.1.4 but I assume this is just a label in the DB 
that I don't need to worry about.

I'm running PHP 5.3.6 on Apache 2.2 on CentOS.
```

Original comment by `supp...@switchomatic.co.uk` on 4 Jul 2011 at 12:22

**Comment by GoogleCodeExporter** [comment]
```
For the record, disabling permalinks does not help.
```

Original comment by `supp...@switchomatic.co.uk` on 4 Jul 2011 at 12:24

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
I've just run into this issue with Wordpress 3.1.4 and CCTM 0.9.3.

All "Pages" simply return a "Not Found" error, and permalinks settings have no 
effect.

The illegal offset type in get_post_type_object() seems to be a red herring. I 
tried adding: 

    if( is_array( $post_type ) )
    {
        $post_type = $post_type[0];
    }

Which causes it to successfully return the correct object.

But the page is still "Not Found"
```

Original comment by `p...@further.co.uk` on 4 Jul 2011 at 1:04

**Comment by GoogleCodeExporter** [comment]
```
Yes, I also tried suppressing the error message but with no fix.

This is a serious problem - essentially if the CCTM plugin is enabled, no 
Wordpress pages (and I presume Posts) can be found.

Incidentally, the advice to disable permalinks doesn't appear to be part of the 
problem - the problem is that the Pages cannot be found when the plugin is 
activated, and Pages should surely not be handled by CCTM at all...
```

Original comment by `supp...@switchomatic.co.uk` on 4 Jul 2011 at 1:08

**Comment by GoogleCodeExporter** [comment]
```
Update:

Traced the error back to archive functionality, and commented out these lines 
in loader.php:

    //add_filter('getarchives_where', 'CCTM::get_archives_where_filter' , 10 , 2);
    //add_filter('request', 'CCTM::request_filter');

I don't need archives on my custom types ( staff profile pages ), so this has 
restored the desired functionality for me, and may also work for you.
```

Original comment by `p...@further.co.uk` on 4 Jul 2011 at 1:09

**Comment by GoogleCodeExporter** [comment]
```
The crux of the issue is, in fact, in CCTM::request_filter. It's checking for 
an empty $query['post_type'] before it does its business. When it's called 
during a request for a page, this value is empty- thus triggering the filter 
and borking the query request for perfectly valid pages.

You can probably get away with just commenting out add_filter('request', 
'CCTM::request_filter'); for now, if you don't use this functionality.
```

Original comment by `p...@further.co.uk` on 4 Jul 2011 at 1:17

**Comment by GoogleCodeExporter** [comment]
```
You legend! Commenting out that line has fixed it on my installation at least. 
I'm now running WP 3.1.4 and CCTM 0.9.3 together and all is working fine. 
Thanks for your help.
```

Original comment by `supp...@switchomatic.co.uk` on 4 Jul 2011 at 1:48

**Comment by GoogleCodeExporter** [comment]
```
Man... good work debugging.  Those other lines re the archives were necessary 
specifically because WP sniped archive functionality: 
http://code.google.com/p/wordpress-custom-content-type-manager/issues/detail?id=
13&can=1&q=archive

What a pain in the ass.  So I'm off to rewrite more of the WP core, it seems.
```

Original comment by `fireproofsocks` on 4 Jul 2011 at 4:04
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
Issue 109 has been merged into this issue.
```

Original comment by `fireproofsocks` on 4 Jul 2011 at 4:11

**Comment by GoogleCodeExporter** [comment]
Original comment by `fireproofsocks` on 4 Jul 2011 at 4:12
- Added labels: **Priority-Critical**
- Removed labels: **Priority-Medium**

**Comment by GoogleCodeExporter** [comment]
```
Fix released as 0.9.3.1
SVN Committed revision 404619.
```

Original comment by `fireproofsocks` on 5 Jul 2011 at 12:35
- Changed state: **Fixed**
